project/stac_api/models.py

```diff
@@ -1,28 +1,307 @@
 from django.contrib.gis.db import models
+from django.utils.translation import gettext_lazy as _
+from django.core.exceptions import ValidationError
+from django.contrib.postgres.fields import ArrayField
+import re
 
 # pylint: disable=fixme
 # TODO remove this pylint disable once this is done
 
 # st_geometry bbox ch
 BBOX_CH = 'POLYGON((2317000 913000,3057000 913000,3057000 1413000,2317000 1413000,2317000 913000))'
 
+# I allowed myself to make excessive use of comments below, as this is still work in progress.
+# all the comments can be deleted later on
+
+# For Collections and Items: No primary key will be defined, so that the auto-generated ones
+# will be used by Django. For assets, a primary key is defined as "PositiveBigInteger" due the
+# expected large number of assets
+
 
 class Collection(models.Model):
-    # TODO: is the service multilingual? title_de, title_fr, title_en, title_it, title_rm
-    bgid_id = models.BigAutoField(primary_key=True)
-    created = models.DateTimeField(auto_now_add=True)
-    updated = models.DateTimeField(auto_now=True)
-
-    layer_id = models.TextField()
-    geocat_uuid = models.UUIDField()
-    goecat_title = models.TextField()
-    geodat_description = models.TextField()
-    gsd = models.TextField()
-    variant = models.TextField()
-    lang = models.TextField()  # TODO: what is its purpose?
-    bbox = models.PolygonField(
-        default=BBOX_CH, srid=2056
-    )  # TODO: directly a geometry for spatial queries? 3D?
-    temporal_interval = models.TimeField(
-    )  # TODO: time field - how many hours are in a (leap) year? textfield?
-    crs = models.URLField(default="http://www.opengis.net/def/crs/OGC/1.3/CRS84")
+    crs = models.URLField(default=["http://www.opengis.net/def/crs/OGC/1.3/CRS84"])  # [string]
+    created = models.DateTimeField(auto_now_add=True)  # datetime
+    updated = models.DateTimeField(auto_now=True)  # datetime
+    description = models.TextField()  # string  / intentionally TextField and
+    # not CharField to provide more space for the text.
+    # TODO: ""description" is required in radiantearth spec, not required in our spec
+
+    # temporal extent will be auto-populated on every item update inside this collection:
+    # start_date will be set to oldest date, end_date to most current date
+    start_date = models.DateTimeField(blank=True)  # will be automatically populated
```

Comment:
it's probably necessary to have `null=True` as well, you have to create a collection before you add any items, in that phase the start_date and end_date will be `null`

Reply:
That's correct, I added null=True